### PR TITLE
Add profile menu

### PR DESCRIPTION
**Pull request by Martino315**

### CODE REVIEW

File: src/components/navigations/app-bar/app-bar.component.js [hunk 1]
```diff
@@ -6,15 +6,48 @@ import IconButton from '@material-ui/core/IconButton';
 import Typography from '@material-ui/core/Typography';
 import Badge from '@material-ui/core/Badge';
 import Hidden from '@material-ui/core/Hidden';
+import Menu from '@material-ui/core/Menu';
+import { withStyles } from '@material-ui/core/styles';
```
Comment (knightburton) [inline src/components/navigations/app-bar/app-bar.component.js]: I would put this at the top of the material imports.

File: src/components/navigations/app-bar/app-bar.component.js [hunk 2]
```diff
@@ -6,15 +6,48 @@ import IconButton from '@material-ui/core/IconButton';
...
+      vertical: 'bottom',
+      horizontal: 'center',
+    }}
+    transformOrigin={{
+      vertical: 'top',
+      horizontal: 'center',
+    }}
+    {...props}
+  />
+));
```
Comment (knightburton) [inline src/components/navigations/app-bar/app-bar.component.js]: This `StyledMenu` is unnecessary, just use the default `Menu` component from core material without any style change.

File: src/components/navigations/app-bar/app-bar.component.js [hunk 3]
```diff
@@ -6,15 +6,48 @@ import IconButton from '@material-ui/core/IconButton';
...
 const AppBar = ({ toggleMobileDrawer }) => {
+  const [anchorEl, setAnchorEl] = React.useState(null);
```
Comment (knightburton) [inline src/components/navigations/app-bar/app-bar.component.js]: I would use a different names to be more straight in the future and alsu I would import the `useState` at the top of the file:
`const [userAnchorEl, setUserAnchorEl] = useState(null);`

File: src/components/navigations/app-bar/app-bar.component.js [hunk 4]
```diff
@@ -6,15 +6,48 @@ import IconButton from '@material-ui/core/IconButton';
...
   const classes = useStyles();
+  function handleClick(event) {
```
Comment (knightburton) [inline src/components/navigations/app-bar/app-bar.component.js]: I would use arrow function for this and I would rename it to be more straight in the future (we will use multiple functions and menus as well here):
```
const handleUserClick = e => setAnchorEl(e.currentTarget);
```

File: src/components/navigations/app-bar/app-bar.component.js [hunk 5]
```diff
@@ -6,15 +6,48 @@ import IconButton from '@material-ui/core/IconButton';
...
+    setAnchorEl(event.currentTarget);
+  }
+
+  function handleClose() {
```
Comment (knightburton) [inline src/components/navigations/app-bar/app-bar.component.js]: Convert this into a more straight arrow function, same reason as before.

File: src/components/navigations/app-bar/app-bar.component.js [hunk 6]
```diff
@@ -53,9 +86,30 @@ const AppBar = ({ toggleMobileDrawer }) => {
           edge="end"
           aria-haspopup="true"
           color="inherit"
+          onClick={handleClick}
         >
           <AccountCircle />
         </IconButton>
+        <StyledMenu
+          id="customized-menu"
```
Comment (knightburton) [inline src/components/navigations/app-bar/app-bar.component.js]: Give this a more meaningful id, like: `user-action-menu`

**Review by knightburton — CHANGES_REQUESTED** [review]

File: .eslintrc.js [hunk 7]
```diff
@@ -30,8 +30,8 @@ module.exports = {
     'max-len': [2, { "code": 175 }],
     'no-console': [1],
     'react/no-array-index-key': 0,
-    'react/prop-types': 1,
+    'react/prop-types': ['error', { ignore: ['classes'] }],
```
Comment (norbertnemeth) [inline .eslintrc.js]: We should use numbers instead of strings.

**Review by norbertnemeth — CHANGES_REQUESTED** [review]

**Review by norbertnemeth — APPROVED** [review]

File: src/components/navigations/app-bar/app-bar.component.js (hunk 1 above)
Comment (norbertnemeth) [reply src/components/navigations/app-bar/app-bar.component.js]: Has been removed

File: src/components/navigations/app-bar/app-bar.component.js (hunk 2 above)
Comment (norbertnemeth) [reply src/components/navigations/app-bar/app-bar.component.js]: StyledMenu  has been removed.

**Review by knightburton — APPROVED** [review]

File: src/components/navigations/app-bar/app-bar.component.js (hunk 3 above)
Comment (Martino315) [reply src/components/navigations/app-bar/app-bar.component.js]: Fixed.

File: src/components/navigations/app-bar/app-bar.component.js (hunk 4 above)
Comment (Martino315) [reply src/components/navigations/app-bar/app-bar.component.js]: Fixed

File: src/components/navigations/app-bar/app-bar.component.js (hunk 5 above)
Comment (Martino315) [reply src/components/navigations/app-bar/app-bar.component.js]: Fixed.

File: src/components/navigations/app-bar/app-bar.component.js (hunk 6 above)
Comment (Martino315) [reply src/components/navigations/app-bar/app-bar.component.js]: StyleMenu is no longer available.

**Pull request merged**